### PR TITLE
Migrate totalSpace to IO Thread

### LINKED ISSUE #4038: Move `totalSpace()` method to IO thread
See parent issue #4035

### PR DESCRIPTION
**Pull request by CalebKL**
Fixes #4038 

This PR, moves `totalSpace` to the IO Thread and removes runBlocking which was blocking the main thread. 

**Screenshots**

### CODE REVIEW

**Comment by kelson42** [comment]
@CalebKL Ready to review?

**Comment by CalebKL** [comment]
> @CalebKL Ready to review?

Yes. I have already talked to  @MohitMaliFtechiz. 
I will proceed with the other tasks from 4035](https://github.com/kiwix/kiwix-android/issues/4035). 
@MohitMaliFtechiz I had accidentally committed the old and new string(tarask) thats why I had to roll back and force-push without the strings. 
Apart from that, the PR is a go for review.

**Comment by MohitMaliFtechiz** [comment]
@CalebKL Thanks, i will review it soon.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/ObjectBoxToRoomMigratorTest.kt [hunk 1]
```diff
@@ -52,7 +52,7 @@ class ObjectBoxToRoomMigratorTest {
   private lateinit var kiwixRoomDatabase: KiwixRoomDatabase
   private lateinit var boxStore: BoxStore
   private lateinit var objectBoxToRoomMigrator: ObjectBoxToRoomMigrator
-  private val migrationMaxTime = 25000
+  private val migrationMaxTime = 100000
```
Comment (MohitMaliFtechiz) [inline app/src/androidTest/java/org/kiwix/kiwixmobile/ObjectBoxToRoomMigratorTest.kt:55]: Why we are increasing this timeout?

**Review by MohitMaliFtechiz — CHANGES_REQUESTED** [review]
@CalebKL Please rebase, and resolve the conflicts, and see the review comments.

**Comment by MohitMaliFtechiz** [comment]
@CalebKL Please resolve the conflicts by rebasing the PR. There are some conflicts due to merge commits.

Screenshot from 2024-12-05 18-58-20

**Review by MohitMaliFtechiz — APPROVED** [review]

**Pull request merged**